File: Reference-Android-Java/app/src/main/java/xyz/fay/reference/networking/response/WeatherResponse.java
package xyz.fay.reference.networking.response;

import xyz.fay.parcel.Parcelable;
import xyz.fay.parcel.Parcelize;

@Parcelize
public final class WeatherResponse extends Parcelable {
    private final WeatherLiveResponse[] lives;
    private final String status;

    public WeatherResponse(WeatherLiveResponse[] lives, String status) {
        this.lives = lives;
        this.status = status;
    }
}

File: Reference-Android-Kotlin/app/src/main/java/xyz/fay/reference/networking/response/WeatherResponse.kt
package xyz.fay.reference.networking.response

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class WeatherResponse(
        val lives: Array<WeatherLiveResponse>,
        val status: String
) : Parcelable

@Parcelize
data class WeatherLiveResponse(
        val city: String,
        val province: String
) : Parcelable
